### PR TITLE
LPFG-729: add audience deadline

### PR DESCRIPTION
**Pull request by jakub-kainos**

### CODE REVIEW

File: src/controllers/module/event/eventController.ts [hunk 1]
```diff
@@ -344,7 +349,7 @@ export class EventController {
 			if (errors.size) {
 				res.render('page/course/module/events/event-location', {
 					eventJson: req.body.eventJson,
-					errors: errors
+					errors: errors,
```
Comment (elxsy) [inline src/controllers/module/event/eventController.ts]: unnecessary comma

File: src/controllers/module/event/eventController.ts [hunk 2]
```diff
@@ -323,7 +328,7 @@ export class EventController {
 		return async (req: Request, res: Response) => {
 			res.render('page/course/module/events/event-location', {
 				event: JSON.parse(req.body.eventJson || '{}'),
-				eventJson: req.body.eventJson
+				eventJson: req.body.eventJson,
```
Comment (elxsy) [inline src/controllers/module/event/eventController.ts]: unnecessary comma

File: src/controllers/module/event/eventController.ts [hunk 3]
```diff
@@ -356,7 +361,9 @@ export class EventController {
 					event
 				)
 				res.redirect(
-					`/content-management/courses/${req.params.courseId}/modules/${req.params.moduleId}/events-overview/${savedEvent.id}`
+					`/content-management/courses/${req.params.courseId}/modules/${
```
Comment (elxsy) [inline src/controllers/module/event/eventController.ts]: please don't spill variable extrapolations over multiple lines

File: src/controllers/module/event/eventController.ts [hunk 4]
```diff
@@ -387,7 +394,7 @@ export class EventController {
 					address: req.body.address,
 					capacity: req.body.capacity,
 					minCapacity: req.body.minCapacity,
-					errors: errors
+					errors: errors,
```
Comment (elxsy) [inline src/controllers/module/event/eventController.ts]: unnecessary comma

File: src/controllers/module/event/eventController.ts [hunk 5]
```diff
@@ -405,17 +412,18 @@ export class EventController {
 					event
 				)
 				res.redirect(
-					`/content-management/courses/${req.params.courseId}/modules/${req.params.moduleId}/events-overview/${req.params.eventId}`
+					`/content-management/courses/${req.params.courseId}/modules/${
```
Comment (elxsy) [inline src/controllers/module/event/eventController.ts]: multiple line again

File: src/lib/dateTime.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import moment = require('moment')
+
+export class DateTime {
+	private static readonly numberToMonthName = require('number-to-date-month-name')
+	private static readonly convert = require('convert-seconds')
+	private static readonly isoRegex = new RegExp(
+		'^(-)?P(?:(\\d+)Y)?(?:(\\d+)M)?(?:(\\d+)D)?' + '(T(?:(\\d+)H)?(?:(\\d+)M)?(?:(\\d+(?:\\.\\d+)?)S)?)?$'
```
Comment (elxsy) [inline src/lib/dateTime.ts]: Moment already has iso range for validation and formatting. Why are we redefining this?

File: src/lib/dateTime.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import moment = require('moment')
```
Comment (elxsy) [inline src/lib/dateTime.ts]: Seems like the implementation of the methods in this library can be achieved very quickly with the moment.

File: src/controllers/module/event/eventController.ts (hunk 1 above)
Comment (jakub-kainos) [reply src/controllers/module/event/eventController.ts]: this is `prettier` formatting

File: src/controllers/module/event/eventController.ts (hunk 2 above)
Comment (jakub-kainos) [reply src/controllers/module/event/eventController.ts]: same as above

File: src/controllers/module/event/eventController.ts (hunk 4 above)
Comment (jakub-kainos) [reply src/controllers/module/event/eventController.ts]: same as above

File: src/controllers/module/event/eventController.ts (hunk 5 above)
Comment (jakub-kainos) [reply src/controllers/module/event/eventController.ts]: `prettier` again

File: src/lib/dateTime.ts (hunk 7 above)
Comment (jakub-kainos) [reply src/lib/dateTime.ts]: agree, I've created a task https://cshrdigitalandanalysis.atlassian.net/browse/LPFG-832 to review date/time code

here I just moved functions to a class for better structure

File: src/lib/dateTime.ts (hunk 6 above)
Comment (jakub-kainos) [reply src/lib/dateTime.ts]: please see comment below - created a task to revisit that part

**Pull request merged**